core/src/test/java/com/google/instrumentation/trace/PropagationUtilTest.java

```diff
@@ -84,180 +87,108 @@ public void propagate_SpanContextNoTracing() {
   }
 
   @Test
-  public void format_SpanContext() {
-    assertThat(PropagationUtil.toHttpHeaderValue(new SpanContext(traceId, spanId, traceOptions)))
-        .isEqualTo("0000000000000000000000000000000061FF0000000000000001340012");
-  }
-
-  @Test(expected = NullPointerException.class)
-  public void toHttpHeaderValue_NullSpanContext() {
-    PropagationUtil.toHttpHeaderValue(null);
-  }
-
-  @Test
   public void toHttpHeaderValue_InvalidSpanContext() {
-    assertThat(PropagationUtil.toHttpHeaderValue(SpanContext.INVALID))
-        .isEqualTo("0000000000000000000000000000000000000000000000000000000000");
+    assertThat(PropagationUtil.toBinaryValue(SpanContext.INVALID))
+        .isEqualTo(
+            new byte[] {
+              0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 2, 0
+            });
   }
 
   @Test
-  public void parseHeaderValue_ValidValues() {
-    assertThat(
-            PropagationUtil.fromHttpHeaderValue(
-                "0000000000000000000000000000000061FF0000000000000001340012"))
+  public void parseHeaderValue_BinaryExampleValue() {
+    assertThat(fromBinaryValue(EXAMPLE_BYTES))
         .isEqualTo(new SpanContext(traceId, spanId, traceOptions));
+    assertThat(fromBinaryValue(EXAMPLE_BYTES).getTraceOptions().getOptions()).isEqualTo(1);
+  }
 
-    assertThat(
-            PropagationUtil.fromHttpHeaderValue(
-                "0000000000000000000000000000000000FF0000000000000001340012"))
-        .isEqualTo(new SpanContext(TraceId.INVALID, spanId, traceOptions));
+  @Test(expected = NullPointerException.class)
+  public void parseBinaryValue_NullInput() throws IOException {
+    PropagationUtil.fromBinaryValue(null);
+  }
 
-    assertThat(
-            PropagationUtil.fromHttpHeaderValue(
-                "0000000000000000000000000000000061000000000000000001340012"))
-        .isEqualTo(new SpanContext(traceId, SpanId.INVALID, traceOptions));
+  @Test(expected = IOException.class)
+  public void parseBinaryValue_EmptyInput() throws IOException {
+    PropagationUtil.fromBinaryValue(new byte[0]);
+  }
 
-    assertThat(
-            PropagationUtil.fromHttpHeaderValue(
-                "0000000000000000000000000000000061FF0000000000000000000000"))
-        .isEqualTo(new SpanContext(traceId, spanId, TraceOptions.DEFAULT));
+  @Test(expected = IOException.class)
+  public void parseBinaryValue_UnsupportedVersionId() throws IOException {
```

Comment:
Should this test check the exception's error message, to make sure that it was thrown for the right reason?